app/src/main/java/edu/uc/it3048/mypokedex/ProfileScreenActivity.kt

```diff
@@ -213,4 +211,21 @@ class ProfileScreenActivity : AppCompatActivity() {
                 Toast.makeText(this, "Error", Toast.LENGTH_LONG).show()
             }
     }
+
+    @SuppressLint("SimpleDateFormat")
+    private fun createImageFile() : File {
+        val timeStamp = SimpleDateFormat("yyyyMMdd_HHmmss".format(Date()))
+        val storageDir : File? = this.getExternalFilesDir(Environment.DIRECTORY_PICTURES)
+        return File.createTempFile("MyPokedex${timeStamp}", ".jpg", storageDir).apply {
+            currentPhotoPath = absolutePath
+        }
+    }
+
+    //endregion
+
+    // Shows list of login options (Email and Google)
+    private fun showSignInOptions(){
+        startActivityForResult(AuthUI.getInstance().createSignInIntentBuilder().setAvailableProviders(loginProviders)
+            .setTheme(R.style.AppTheme).build(), LOGIN_REQUEST_CODE)
+    }
 }
```

Comment:
most of the giant blocks of red and green are just organizing the functions into the respective regions.